File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Type def - Class 04 - Interface impl.cs
using System;

public class Class1 : IDisposable
{
    public void Dispose()
    {
    }
}

public class Class2 : IDisposable
{
    void IDisposable.Dispose()
    {
    }
}

public class Class3 : IDisposable
{
}

File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Type def - Class 04 - Interface impl.fs
module Module

open System

do
    use c1 = new Class1()
    c1.Dispose()
    (c1 :> IDisposable).Dispose()
    
    use c2 = new Class2()
    c2.Dispose()
    (c2 :> IDisposable).Dispose()

    use c3 = new Class3()
    c3.Dispose()
    (c3 :> IDisposable).Dispose()
